### PR TITLE
py: Support zip builtin feature

### PR DESCRIPTION
**Pull request by corona10**
Implement built-in `zip`

### CODE REVIEW

**Comment by corona10** [comment]
@ncw Please take a look! Thanks alot!

File: py/zip.go [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+	itTuple Tuple
+	size    int
+}
+
+// A python ZipIterator iterator
+type ZipIterator struct {
+	zip Zip
+}
+
+var ZipType = NewTypeX("zip", `zip(seq1 [, seq2 [...]]) -> [(seq1[0], seq2[0] ...), (...)]
```
[the remark this answers is not in the document]
Comment (corona10) [reply py/zip.go]: Done

File: py/tests/zip.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# Copyright 2018 The go-python Authors.  All rights reserved.
+# Use of this source code is governed by a BSD-style
+# license that can be found in the LICENSE file.
+
+doc="zip"
+a = [1, 2, 3, 4, 5]
+b = [10, 11, 12]
+c = [e for e in zip(a, b)]
+for idx, e in enumerate(c):
```
Comment (ncw) [inline py/tests/zip.py]: I think this could do with an

    assert len(c) == 3

**Comment by corona10** [comment]
@ncw 
Done, Thank you for your review.
I hope your family's health improves now.
While I served a military service, one of my comrades was sick with norovirus and he told me that that time was one of the worst experience in his life.

**Comment by corona10** [comment]
And always thank you for always giving me a kind review even though you are busy.

File: py/tests/zip.py [hunk 3]
```diff
@@ -0,0 +1,10 @@
+# Copyright 2018 The go-python Authors.  All rights reserved.
+# Use of this source code is governed by a BSD-style
+# license that can be found in the LICENSE file.
+
+doc="zip"
+a = [1, 2, 3, 4, 5]
+b = [10, 11, 12]
+c = [e for e in zip(a, b)]
+assert len(c) == 3
```
Comment (ncw) [inline py/tests/zip.py]: Where has the check of the contents gone? :-)

Comment (corona10) [reply py/tests/zip.py]: Oh, I've just understood that replace it by inline.. :)
I will revert the part :)

Comment (corona10) [reply py/tests/zip.py]: @ncw Done!

**Comment by ncw** [comment]
That looks great now - thank you :-)

**Pull request merged**